src/gurobi_optimods/metromap.py

```diff
@@ -0,0 +1,1242 @@
+     considering the origin geographical data if available given as geographic position of the nodes
+
+    Parameters
+    ----------
+    graph : Graph
+        networkx graph with node attribute 'pos' being a tuple of x- and y-coordinate
+    linepath_data : DataFrame
+        DataFrame with information on the line routes/paths.
+        It must include "linename", "edge_source", and "edge_target".
+        This data frame could also be empty
+    include_planarity : bool
+        parameter to turn off the consideration of planarity constraints
+    penalty_edge_directions : float
+        weight for original direction part in the objective, default is 1
+    penalty_line_bends : float
+        weight for line-bend part in the objective, default is 1
+    penalty_distance : float
+        weight for distance part in the objective, default is 1
+
+    Returns
+    -------
+    tuple
+        - networkx graph with node property pos_oct representing the x and y coordinates of the octilinear representation
+        - Python dict for direction for each edge (needed for plotting)
+    """
+    if nx is None:
+        raise RuntimeError("Mod needs networkx package!")
+
+    missing_data = False
+    if linepath_data is not None:
+        if "edge_source" not in linepath_data.columns:
+            logger.info("column edge_source not present in linepath_data")
+            missing_data = True
+        if "edge_target" not in linepath_data.columns:
+            logger.info("column edge_target not present in linepath_data")
+            missing_data = True
+        if linepath_data.isnull().values.any():
+            logger.info("some value is nan or format is not correct for linepath_data")
+            missing_data = True
+    if missing_data:
+        raise ValueError("Cannot run optimization. Some data is wrong or missing!")
+
+    # octilinear graph representation needs a maximum node degree of 8
+    # check node degree and abort if infeasible
+    for v in graph.nodes:
+        if graph.degree(v) > 8:
+            logger.info(
+                f"Node with number {v} has node degree {len(graph.edges(v))}. "
+                f"Octilinear representation is not possible for node degree larger than 8"
+            )
+            return  ## value ???
+
+    if linepath_data is None:
+        linepaths = pd.DataFrame()
+    else:
+        linepaths = (
+            linepath_data.set_index("linename")
+            .groupby(["linename"])
+            .apply(
+                lambda x: [(k, v) for k, v in zip(x["edge_source"], x["edge_target"])]
+            )
+        )
+
+    # check whether graph contains node position attribute
+    pos = nx.get_node_attributes(graph, "pos")
+    if len(pos) > 0:
+        geodata = True
+    else:
+        geodata = False
+
+    # remove isolated nodes
+    graph.remove_nodes_from(list(nx.isolates(graph)))
+
+    # restrict parameter to the interval of [0,100]
+    if penalty_edge_directions < 0:
```

Comment:
I would maybe `logger.warn` here to make sure the user knows their settings are being reset